convert2rhel/actions/post_conversion/kernel_boot_files.py

```diff
@@ -0,0 +1,96 @@
+
+        # For Oracle/CentOS Linux 8 the `kernel` is just a meta package, instead,
+        # we check for `kernel-core`. This is not true regarding the 7.* releases.
+        kernel_name = "kernel-core" if system_info.version.major >= 8 else "kernel"
+
+        # Either the package is returned or not. The return_code will be 0 in
+        # either case, so we don't care about checking for that here.
+        output, _ = run_subprocess(["rpm", "-q", "--last", kernel_name], print_output=False)
+
+        # We are parsing the latest kernel installed on the system, which at this
+        # point, should be a RHEL kernel. Since we can't get the kernel version
+        # from `uname -r`, as it requires a reboot in order to take place, we are
+        # detecting the latest kernel by using `rpm` and figuring out which was the
+        # latest kernel installed.
+        latest_installed_kernel = output.split("\n")[0].split(" ")[0]
+        latest_installed_kernel = latest_installed_kernel[len(kernel_name + "-") :]
+        grub2_config_file = grub.get_grub_config_file()
+        initramfs_file = INITRAMFS_FILEPATH % latest_installed_kernel
+        vmlinuz_file = VMLINUZ_FILEPATH % latest_installed_kernel
+
+        logger.info("Checking if the '%s' file exists.", vmlinuz_file)
+        vmlinuz_exists = os.path.exists(vmlinuz_file)
+        logger.info("Checking if the '%s' file exists.", initramfs_file)
+        is_initramfs_valid = checks.is_initramfs_file_valid(initramfs_file)
+
+        if is_initramfs_valid or vmlinuz_exists:
+            logger.info("The initramfs and vmlinuz files are valid.")
+            return
+
+        logger.warning(
+            "Couldn't verify the kernel boot files in the boot partition. This may cause problems during the next boot "
+            "of your system.\nIn order to fix this problem you may need to free/increase space in your boot partition"
+            " and then run the following commands in your terminal:\n"
+            "1. yum reinstall %s-%s -y\n"
+            "2. grub2-mkconfig -o %s\n"
+            "3. reboot",
+            kernel_name,
+            latest_installed_kernel,
+            grub2_config_file,
+        )
+        self.add_message(
+            level="WARNING",
+            id="UNABLE_TO_VERIFY_KERNEL_BOOT_FILES",
+            title="Unable to verify kernel boot files",
+            description="Couldn't verify the kernel boot files in the boot partition. This may cause problems during the next boot "
+            "of your system.",
+            remediations=(
+                "In order to fix this problem you may need to free/increase space in your boot partition"
+                " and then run the following commands in your terminal:\n"
+                "1. yum reinstall %s-%s -y\n"
+                "2. grub2-mkconfig -o %s\n"
+                "3. reboot" % (kernel_name, latest_installed_kernel, grub2_config_file)
+            ),
+        )
```

Comment:
```suggestion
        remediations = (
                "In order to fix this problem you may need to free/increase space in your boot partition"
                " and then run the following commands in your terminal:\n"
                "1. yum reinstall %s-%s -y\n"
                "2. grub2-mkconfig -o %s\n"
                "3. reboot" % (kernel_name, latest_installed_kernel, grub2_config_file)
            )
        logger.warning(
            "Couldn't verify the kernel boot files in the boot partition. This may cause problems during the next boot "
            "of your system.\n%s", remediations
        )
        self.add_message(
            level="WARNING",
            id="UNABLE_TO_VERIFY_KERNEL_BOOT_FILES",
            title="Unable to verify kernel boot files",
            description="Couldn't verify the kernel boot files in the boot partition. This may cause problems during the next boot "
            "of your system.",
            remediations=remediations,
        )
```